novawallet/Common/Services/ChainRegistry/ChainRegistry.swift

```diff
@@ -293,11 +310,20 @@ extension ChainRegistry: ChainRegistryProtocol {
     func chainsSubscribe(
         _ target: AnyObject,
         runningInQueue: DispatchQueue,
+        filterStrategy: ChainFilterStrategy,
         updateClosure: @escaping ([DataProviderChange<ChainModel>]) -> Void
     ) {
+        var currentChains = availableChains
+
         let updateClosure: ([DataProviderChange<ChainModel>]) -> Void = { changes in
             runningInQueue.async {
-                updateClosure(changes)
+                let filtered = filterStrategy.filter(
+                    changes,
+                    using: currentChains
+                )
+                currentChains = filtered.mergeToDict(currentChains)
```

Comment:
I think it will not work. Imagine a chain became disabled - then the filtered changes will contain .delete(...) change that will delete it from ```currentChains``` on merge. So, we need to merge not filtered but original changes I think. 

Only concern i have in this implementation is that we duplicate chains storage across all the subscriptions. I think better solution will be to store subscribers with filters directly in ChainRegistry and notify them in the end of the ```handle(changes:...)``` method. In this case we reuse existing availableChains storage for all subscribers and also guarantee that once they got chain runtimeProvider and connection is also in place to be queried - currently, we don't guarantee that.